parsers/yacc.c

```diff
@@ -102,22 +102,20 @@ static bool leave_union (const char *line CTAGS_ATTR_UNUSED,
 
 static void make_promise_for_epilogue (void)
 {
-	const unsigned char *tmp, *last;
+	const unsigned char *tmp;
 	long endCharOffset;
 	unsigned long c_start;
 	unsigned long c_source_start;
 	unsigned long c_end;
 
-	c_start = getInputLineNumber ();
-	c_source_start = getSourceLineNumber();
+	/* We are at line with %% so the next line is the start of epilogue */
+	c_start = getInputLineNumber () + 1;
+	c_source_start = getSourceLineNumber() + 1;
 
 	/* Skip the lines for finding the EOF. */
 	endCharOffset = 0;
-	last = NULL;
 	while ((tmp = readLineFromInputFile ()))
-		last = tmp;
-	if (last)
-		endCharOffset = strlen ((const char *)last);
+		endCharOffset = strlen ((const char *)tmp);
```

Comment:
The trickyness of that loop might warrant a comment for nobody to come by it and "improve" it like it was before.